### DISCUSSION TITLE
Run frosting commands without TaskNameAttribute

### DISCUSSION THREAD

**Discussion by lskyum** [cake-build/cake#3444] [Q&A] [answered]
Thanks for this awesome library!

Is there a way to run commands like DotNetCorePublish, with Cake.Frosting, but without using FrostingTask and TaskNameAttribute?

I want to use the commands from ICakeContext, in an existing console application, but without managing the flow of the application with cake tasks. It is possible?

**Comment by pascalberger** [comment, marked as answer]
Aliases, like `DotNetCorePublish`, are defined in `Cake.Common` and independent of the runner (Frosting, .NET Tool, etc). 

There is [Cake.Bridge](https://github.com/devlead/Cake.Bridge) which allows you to use functionality from Cake.Core / Cake.Common from any .NET project language.

**Reply by devlead** [reply]
There's also [Cake Bridge Dependency Injection](https://www.devlead.se/posts/2021/2021-01-12-introducing-cake-bridge-dependency-injection) if you're using DI.

**Reply by lskyum** [reply]
Thanks to you both @pascalberger and @devlead!

Cake.Bridge.DependencyInjection looks exactly like what I need, but when I run the Hello World example I get an error:
_Method 'GetArguments' in type 'Cake.Bridge.DependencyInjection.BridgeArguments' from assembly 'Cake.Bridge.DependencyInjection, Version=0.5.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation._

Any idea how to fix that?

**Reply by augustoproiete** [reply]
@lskyum Cake v1.2.0 was released yesterday and includes a change that requires `Cake.Bridge.DependencyInjection` to be updated to target Cake v1.2.0 as well. Tracked via https://github.com/devlead/Cake.Bridge.DependencyInjection/issues/1

In the meantime, a possible workaround for you is to target Cake v1.1.0 instead

```xml
  <ItemGroup>
    <PackageReference Include="Cake.Common" Version="1.1.0" />
    <PackageReference Include="Cake.Bridge.DependencyInjection" Version="0.5.0" />
  </ItemGroup>
```

**Reply by lskyum** [reply]
@augustoproiete Cake v1.1.0 works, thank you!

**Reply by devlead** [reply]
Cake.Bridge.DependencyInjection 0.6.0 is now released and it supports 1.2.0.
https://www.nuget.org/packages/Cake.Bridge.DependencyInjection/0.6.0

**Comment by pitermarx** [comment]
Not sure if this might interest you
https://github.com/pitermarx/Cake.Console

but you can do this
```cs
using Cake.Console;
using Cake.Common.Tools.DotNetCore;

var cake = new CakeHostBuilder(args).Build();
cake.Context.DotNetCorePublish("your.csproj");
```

**Reply by lskyum** [reply]
@pitermarx Thanks for the suggestion! That also looks like a simple and good solution. I have Cake.Bridge.DependencyInjection working now, but may also give this a try.
